### PR TITLE
Add optional format selector for YAML or JSON

### PR DESCRIPTION
**Pull request by martinb3**
- Add new CLI option `-f` or `--format`, legal values are `yaml` or `json`, default to `json`
- Add cfn_flip to produce YAML from JSON, currently recommended way (RE: https://github.com/cloudtools/troposphere/issues/567)
- Add conditional for the new CLI option; tested & confirmed YAML is produced

### CODE REVIEW

**Comment by martinb3** [comment]
I've built this branch to check out what YAML versions of our own templates look like; still defaults to json, but I found the exercise interesting. I'm convinced `cfn_flip` is the best way to do it, still, based on the output.

File: jetstream/cli.py [hunk 1]
```diff
@@ -140,6 +143,9 @@ def main():
     parser.add_argument('--public', action='store_true',
                         help='Whether S3 Published documents should be public',
                         default=False)
+    parser.add_argument('--format', '-f', dest='format',
```
Comment (cwgem) [inline jetstream/cli.py]: Should validate the argument via `choices` 

http://stackoverflow.com/questions/15836713/allowing-a-specific-value-for-an-argparse-argument

File: jetstream/template.py [hunk 2]
```diff
@@ -254,12 +255,21 @@ def generate(self, testing=False):
...
         try:
-            # Handle JSON.dumps failing
-            encoded_template = json.dumps(
-                tmpl.to_dict(),
-                sort_keys=False, indent=2,
-                separators=(',', ': '),
-                cls=JetstreamEncoder)
+            if fmt == 'json':
+                # Handle JSON.dumps failing
+                encoded_template = json.dumps(
```
Comment (cwgem) [inline jetstream/template.py]: If we're going to do the flip method against JSON, I'd rather the JSON dump be a centralized variable so if we need to update the JSON parameters it only happens in one place. Unless the plan is that the JSON options might be different when doing a specific flip.

File: jetstream/cli.py (hunk 1 above)
Comment (martinb3) [inline jetstream/cli.py]: Fixed 👍

File: jetstream/template.py (hunk 2 above)
Comment (martinb3) [inline jetstream/template.py]: Fixed 👍

**Pull request merged**